poop.ts

```diff
@@ -0,0 +1,71 @@
+import { Server } from "./mod.ts";
+import { WebSocketClient } from "./src/websocket_client.ts";
+
+const server: Server = new Server();
+
+// Run the server
+server.runWs({
+  hostname: "127.0.0.1",
+  port: 5001,
+});
+
+console.log(
+  `Server started on ws://${server.hostname}:${server.port}`,
+);
+
+// Example using the connect handler
+server.on("connect", (e: CustomEvent) => {
+  const { id } = e.detail
+  console.log('server had a client connected: ' + id)
+})
+
+// Example using the disconnect handler
+server.on("disconnect", (e: CustomEvent) => {
+  const { id, code, reason } = e.detail
+  console.log('server had a client disconnect:')
+  console.log(id, code, reason)
+})
+
+// Example using generics + custom channel handler and getting the data from the event
+type UserMessage = {
+  username: string;
+  sender: number;
+  id: number // client socket id
+};
+server.on("channel", (event: CustomEvent<UserMessage>) => {
+  console.log('got channel msg')
+  const { username, sender, id } = event.detail
+  
+  // Example sending to all clients, specifying which channel it is for
+  server.to("chat-message", {
+    message: "yep, the server got your msg :)"
+  })
+
+  // example sending to a specific client TODO :: Pass socket id as first param?
+  server.to("chat-message", {
+    message: "this message is only for you *wink wink*"
+  }, {
+    id,
+    only: true
+  })
+
+  // example sending to all OTHER clients TODO :: broadcast?
+  server.to("chat-message", {
+    message: "You got this message but client with id of " + id + " shouldnt have"

```

Comment:
yea this would be a broadcast

Reply:
https://github.com/drashland/wocket/pull/124/commits/13dfc572c46016def7f1c0cd46f085bf06489c2e